dockstore-webservice/src/main/java/io/dockstore/webservice/core/CollectionEntry.java

```diff
@@ -18,32 +19,48 @@
     private long id;
     private String entryType;
     private String versionName;
+    private DescriptorLanguage descriptorType = null;
```

Comment:
I think this needs to be a List, doesn't it? To handle tools?

Reply:
Yes for tools it should be a list, this is what I'm currently working on. The `descriptorType` you see here is for workflows, which should be an enum

Reply:
> Yes for tools it should be a list, this is what I'm currently working on. The `descriptorType` you see here is for workflows, which should be an enum

Is your goal to have a different CollectionEntry subclass for tools?

Reply:
Might be easier to just have it as a list.  Workflows can just have a single element in the list

Reply:
> Might be easier to just have it as a list. Workflows can just have a single element in the list

Agreed.